### PR TITLE
Updated to Symfony 3.2.1 final

### PR DESCRIPTION
**Pull request by Guite**
| Q                 | A
| ----------------- | ---
| Bug fix?          | no
| New feature?      | no
| BC breaks?        | no
| Deprecations?     | no
| Fixed tickets     | #3027, #2099, #2639
| Refs tickets      | -
| License           | MIT
| Changelog updated | yes

### CODE REVIEW

File: src/lib/Zikula/Bundle/FormExtensionBundle/Form/Type/LocaleType.php [hunk 1]
```diff
@@ -29,13 +30,13 @@ public function setDefaultOptions(OptionsResolverInterface $resolver)
         ]);
     }
 
-    public function getName()
+    public function getBlockPrefix()
```
Comment (craigh) [inline src/lib/Zikula/Bundle/FormExtensionBundle/Form/Type/LocaleType.php]: I believe this change was introduced in Symfony 2.8. If that is the case here (or with other things you are changing) then they should be changed only in the Core-1.4 branch and then merged up to Core-2.0. This means that large monolithic changes like this PR (currently) are probably not the best way to handle this transition.

ANYTHING that will work with Symfony 2.8 must be first merged into Core-1.4 and then merged up.

File: src/lib/Zikula/Core/Forms/ZikulaCsrfProvider.php [hunk 2]
```diff
@@ -11,12 +11,12 @@
...
 namespace Zikula\Core\Forms;
 
-use Symfony\Component\Form\Extension\Csrf\CsrfProvider\CsrfProviderInterface;
+use Symfony\Component\Security\Csrf\CsrfTokenManagerInterface;
 
 /**
  * Csrf provider based on zikula csrf system.
  */
-class ZikulaCsrfProvider implements CsrfProviderInterface
+class ZikulaCsrfProvider implements CsrfTokenManagerInterface
```
Comment (craigh) [inline src/lib/Zikula/Core/Forms/ZikulaCsrfProvider.php]: I would be curious to know if the `ZikulaCsrfProvider` is even necessary at all. Why aren't we using the native one?

File: src/lib/util/ModUtil.php [hunk 3]
```diff
@@ -11,7 +11,7 @@
 
 use Symfony\Bundle\FrameworkBundle\Controller\ControllerNameParser;
 use Symfony\Bundle\FrameworkBundle\Controller\ControllerResolver;
-use Symfony\Component\DependencyInjection\ContainerAware;
+use Symfony\Component\DependencyInjection\ContainerAwareInterface;
 use Zikula\Core\Event\GenericEvent;
```
Comment (craigh) [inline src/lib/util/ModUtil.php]: changes to ModUtil (or any Util class) in the master branch are a waste of time. They are **all** going to be deleted.

File: src/system/ThemeModule/Resources/config/services.xml [hunk 4]
```diff
@@ -23,9 +23,8 @@
         </service>
 
         <service id="zikula_theme_module.theme_entity.repository"
-                 class="%zikula_theme_module.theme_entity.repository.class%"
-                 factory-service="doctrine"
-                 factory-method="getRepository">
+                 class="%zikula_theme_module.theme_entity.repository.class%">
+            <factory service="doctrine" method="getRepository" />
```
Comment (craigh) [inline src/system/ThemeModule/Resources/config/services.xml]: can this be done in Symfony 2.8?

File: src/lib/Zikula/Core/Forms/ZikulaCsrfProvider.php (hunk 2 above)
Comment (Guite) [inline src/lib/Zikula/Core/Forms/ZikulaCsrfProvider.php]: If we switch to the native one the custom implementation should be removed from 1.4 and 2.0-spec then. But as part of a separate issue.

File: src/lib/Zikula/Bundle/FormExtensionBundle/Form/Type/LocaleType.php (hunk 1 above)
Comment (Guite) [inline src/lib/Zikula/Bundle/FormExtensionBundle/Form/Type/LocaleType.php]: See #3202

File: src/system/ThemeModule/Resources/config/services.xml (hunk 4 above)
Comment (Guite) [inline src/system/ThemeModule/Resources/config/services.xml]: See #3202

**Pull request merged**